llvm/lib/Analysis/InstructionSimplify.cpp

```diff
@@ -4635,33 +4635,29 @@ static Value *simplifySelectWithICmpEq(Value *CmpLHS, Value *CmpRHS,
 
 /// Try to simplify a select instruction when its condition operand is an
 /// integer comparison.
-static Value *simplifySelectWithICmpCond(Value *CondVal, Value *TrueVal,
-                                         Value *FalseVal,
-                                         const SimplifyQuery &Q,
-                                         unsigned MaxRecurse) {
+static Value *simplifySelectWithCmpCond(Value *CondVal, Value *TrueVal,
```

Comment:
I don't think this function should change, nearly all code in here is not relevant for fcmp. Add a call to simplifySelectWithEquivalence in simplifySelectWithFCmp instead.

Reply:
Wouldn't there be code duplication then? What are your thoughts on absorbing simplifySelectWithFCmp into this function?

Reply:
I don't think there is any substantial amount of overlap between the functions. Note that while this function contains a lot of calls to simplifySelectWithICmpEq, only the plain variant of them is relevant to the FP case. The ones involving and/or are integer-only.